### ISSUE TITLE
hope styled-components support decorators style 

### ISSUE THREAD

**Issue by roadhub** [styled-components/styled-components#1347]
## Version
styled-components: 2.2.4
babel-plugin-styled-components: 1.3.0

## Reproduction

https://www.webpackbin.com/bins/-KeeZCr0xKfutOfOujxN
https://codesandbox.io/s/rkmNRByE4

## Actual Behavior

```javascript
let style = `
     background:  red;
`
@styled(style)
class component extends React.component{

    render(){}
}
```

**Comment by philpl** [comment]
No we don't support the HOC pattern where the styles would come first two very simple reasons:

1. We don't need to implement it because you could implement it yourself quite easily
2. It's not a pattern we recommend

So regarding 2), we don't recommend wrapping components in styled components. Rather you should try to wrap styled components, which usually are primitives and just replace your elements, in your components.

This follows the pattern of: containers/structural components -> presentational components, where the latter are styled components. This pattern scales a lot better than mixing this and disrupting the usual order.

In short this we do not recommend: containers/structural components -> presentational components -> containers/structural components -> presentational components
